extensions/parquet/table/src/main/java/io/deephaven/parquet/table/ParquetTableWriter.java

```diff
@@ -90,13 +90,37 @@ enum CacheTags {
      * @param <B>
      */
     interface TransferObject<B> extends SafeCloseable {
-        void propagateChunkData();
+        /**
+         * Fetch all data corresponding to the provided row sequence.
+         */
+        void fetchData(RowSequence rs);
 
-        B getBuffer();
+        /**
+         * Copy all the fetched data into an internal buffer, which can then be accessed using
+         * {@link TransferObject#getBuffer()}. This method should only be called after
+         * {@link TransferObject#fetchData(RowSequence)}}. Note that this method can lead to Out-of-Memory error for
+         * types like strings if the fetched data is too big.
+         *
+         * @return The number of fetched data entries copied into the buffer.
+         */
+        int bufferAllFetchedData();
```

Comment:
I recommend a clearer name:
transferAllToBuffer
And a re-write to the JavaDoc to reflect that this is meant to be used when writing unpaginated data, and not meant to be interleaved with calls to the other method.